pkg/rbac/sharedData.go

```diff
@@ -32,116 +40,173 @@ type resource struct {
+		shared.csResources = []resource{}
+		return shared.csErr
+	}
+
+	if rows != nil {
+		defer rows.Close()
+
+		for rows.Next() {
+			var kind string
+			var apigroup string
+			err := rows.Scan(&apigroup, &kind)
+			if err != nil {
+				klog.Errorf("Error %s retrieving rows for query:%s for apigroup %s and kind %s", err.Error(), query,
+					apigroup, kind)
+				continue
+			}
+
+			shared.csResources = append(shared.csResources, resource{apigroup: apigroup, kind: kind})
 
-		rows, queryerr := cache.pool.Query(ctx, query)
-		if queryerr != nil {
-			klog.Errorf("Error resolving query [%s]. Error: [%+v]", query, queryerr.Error())
-			shared.csErr = queryerr
-			shared.csResources = []resource{}
-			return shared.csResources, shared.csErr
 		}
+	}
+	shared.csUpdatedAt = time.Now()
 
-		if rows != nil {
-			defer rows.Close()
+	return shared.csErr
+}
 
-			for rows.Next() {
-				var kind string
-				var apigroup string
-				err := rows.Scan(&apigroup, &kind)
-				if err != nil {
-					klog.Errorf("Error %s retrieving rows for query:%s for apigroup %s and kind %s", err.Error(), query,
-						apigroup, kind)
-					continue
-				}
+// Obtain all the namespaces in the hub cluster.
+// Equivalent to `oc get namespaces`
+func (shared *SharedData) GetSharedNamespaces(cache *Cache, ctx context.Context) error {
+	shared.nsLock.Lock()
+	defer shared.nsLock.Unlock()
+	//empty previous cache
+	shared.namespaces = nil
+	shared.nsErr = nil
 
-				shared.csResources = append(shared.csResources, resource{apigroup: apigroup, kind: kind})
+	klog.V(5).Info("Getting namespaces from Kube Client..")
 
-			}
-		}
+	clientset, kubeErr := kubernetes.NewForConfig(cache.restConfig)
+	if kubeErr != nil {
+		klog.Warning("Error with creating a new clientset.", kubeErr.Error())
+		shared.nsErr = kubeErr
+		shared.nsUpdatedAt = time.Now()
+		return shared.nsErr
+	}
 
-		// get all namespaces in cluster and cache in shared.namespaces.
-		shared.nsLock.Lock()
-		defer shared.nsLock.Unlock()
-		allNamespaces, nsErr := shared.GetSharedNamespaces(cache, ctx)
-		if nsErr != nil {
-			shared.nsErr = nsErr
-		}
+	namespaceList, nsErr := clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if nsErr != nil {
+		klog.Warning("Error resolving namespaces from KubeClient: ", nsErr)
+		shared.nsErr = nsErr
+		shared.nsUpdatedAt = time.Now()
+		return shared.nsErr
+	}
 
-		// update the cache.
-		shared.namespaces = allNamespaces
-		shared.csUpdatedAt = time.Now()
-	} else {
-		klog.V(8).Info("Using cluster scoped resources from cache.")
+	// add namespaces to allNamespace List
+	for _, n := range namespaceList.Items {
+		shared.namespaces = append(shared.namespaces, n.Name)
 	}
+	shared.nsUpdatedAt = time.Now()
 
-	return shared.csResources, shared.csErr
+	return shared.nsErr
 }
 
-func (shared *SharedData) GetSharedNamespaces(cache *Cache, ctx context.Context) ([]string, error) {
+func (shared *SharedData) GetSharedManagedClusters(cache *Cache, ctx context.Context) error {
```

Comment:
No need to use `Shared` in the function name.
```suggestion
func (shared *SharedData) GetManagedClusters(cache *Cache, ctx context.Context) error {
```